### PR TITLE
Document the latest metrics improvements

### PR DESCRIPTION
**Pull request by irevoire**
🤖  [API Diff]() _Put the link of the GitHub comment generated by bump.sh if generated; Apply the `OpenApi` label_

---

# Summary

Implemented in https://github.com/meilisearch/meilisearch/pull/3789

- Prefix all the metrics by `meilisearch` so they're easier to find
- Add the real database size used by meilisearch
- Add metrics on the task queue
- Add the `meilisearch_last_update` and `meilisearch_is_indexing` introduced in https://github.com/meilisearch/meilisearch/issues/3843

### CODE REVIEW

File: text/0174-metrics-api.md [hunk 1]
```diff
@@ -51,7 +53,7 @@ Returns the number of times an API resource is accessed.
 http_requests_total{method=":httpMethod",path=":resourcePath"} :numberOfRequest
```
Comment (gmourier) [inline text/0174-metrics-api.md:53]:
```suggestion
meilisearch_http_requests_total{method=":httpMethod",path=":resourcePath"} :numberOfRequest
```

File: text/0174-metrics-api.md [hunk 2]
```diff
@@ -77,7 +79,7 @@ http_response_time_seconds_sum{method=":httpMethod",path=":resourcePath"} :numbe
 http_response_time_seconds_count{method=":httpMethod",path=":resourcePath"} :numberOfRequest
```
Comment (gmourier) [inline text/0174-metrics-api.md:79]:
```suggestion
meilisearch_http_response_time_seconds_count{method=":httpMethod",path=":resourcePath"} :numberOfRequest
```

File: text/0174-metrics-api.md [hunk 3]
```diff
@@ -77,7 +79,7 @@ http_response_time_seconds_sum{method=":httpMethod",path=":resourcePath"} :numbe
...
 ```
 
-#### 3.2.3. `meilisearch_database_size_bytes`
+#### 3.2.3. `meilisearch_db_size_bytes`
 
 Returns the size of the database in bytes.
```
Comment (gmourier) [inline text/0174-metrics-api.md:84]: Could we add some precision for that one? Saying it represents the memory-mapped space "taken" by LMDB is okay to me!

**Review by gmourier — CHANGES_REQUESTED** [review]
Thanks @irevoire 🚂

File: text/0174-metrics-api.md (hunk 1 above)
Comment (irevoire) [reply text/0174-metrics-api.md:53]: Oops thanks

**Review by gmourier — COMMENTED** [review]
@irevoire I just realized we have a `sampleResponse` field listing _some_ metrics values under the Open API /metrics endpoint; we could either remove that sample response or update the example by including the `meilisearch_` prefix. It's up to you!

File: text/0174-metrics-api.md [hunk 4]
```diff
@@ -107,6 +122,24 @@ Returns the total number of index for the Meilisearch instance.
 meilisearch_index_count :numberOfIndex
 ````
 
+#### 3.2.7. `meilisearch_nb_tasks`
+
+Returns the total number of index for the Meilisearch instance.
```
Comment (94noni) [inline text/0174-metrics-api.md:127]: hey, small typo found passing by reading more on meilisearch specs repo
seems a copy past of the above metric name desc

**Review by macraig — APPROVED** [review]

File: open-api.yaml [hunk 5]
```diff
@@ -3885,37 +3885,75 @@ paths:
...
-                  http_response_time_seconds_bucket{method="GET",path="/metrics",le="0.0012"} 1
-                  http_response_time_seconds_bucket{method="GET",path="/metrics",le="0.0015"} 1
-                  http_response_time_seconds_bucket{method="GET",path="/metrics",le="0.002"} 1
-                  http_response_time_seconds_bucket{method="GET",path="/metrics",le="0.003"} 1
-                  http_response_time_seconds_bucket{method="GET",path="/metrics",le="1"} 2
-                  http_response_time_seconds_bucket{method="GET",path="/metrics",le="+Inf"} 2
-                  http_response_time_seconds_sum{method="GET",path="/metrics"} 0.0056515409999999995
-                  http_response_time_seconds_count{method="GET",path="/metrics"} 2
-                  # HELP meilisearch_db_size_bytes Meilisearch Db Size In Bytes
+                  http_response_time_seconds_bucket{method="GET",path="/metrics",le="0"} 0
```
Comment (gmourier) [inline open-api.yaml:3890]:
```suggestion
                  meilisearch_http_response_time_seconds_bucket{method="GET",path="/metrics",le="0"} 0
```

**Review by gmourier — CHANGES_REQUESTED** [review]
Should it be prefixed by `meilisearch` on every metrics?

**Review by gmourier — APPROVED** [review]

**Pull request merged**